inject/maven-plugin/README.md

```diff
@@ -54,9 +56,9 @@ example usage:
+Also note that Helidon Injection strives to ensure your application stays as deterministic as possible (as shown by the <i>Injection__Application/i> generated class). But when the <i>jakarta.inject.Provider</i> type is used within your application then some of that deterministic behavior goes away. This is due to how Provider<>'s work since the implementation for the Provider (i.e., your application logic) "owns" the behavior for what actual concrete type that are created by it, along with the scope/cardinality for those instances. These instances are then delivered (as potentially injectable services) into other dependent services.  In this way Helidon Injection is simply acting as a broker and delivery mechanism between your Provider<> implementation(s) and the consumer that are using those service instances as injection points, etc. This is not meant to scare ore even dissuade you from using Provider<>, but merely to inform you that some of the deterministic behavior goes away under these circumstances using Provider instead of another Scope type like @Singleton. In many/most cases this is completely normal and acceptable. As a precaution, however, Helidon Injection chose to fail-fast at application creation time if your application is found to use jakarta.inject.Provider<T>. You will then need to provide a strategy/configuration in your pom.xml file to permit these types of usages. There are options to allow ALL providers (as shown in the above example), or the strategy can be dictated on a case-by-case basis. See the javadoc for [AbstractApplicationCreatorMojo](src/main/java/io/helidon/inject/maven/plugin/AbstractApplicationCreatorMojo.java) for details.
```

Comment:
```suggestion
Also note that Helidon Injection strives to ensure your application stays as deterministic as possible (as shown by the <i>Injection__Application/i> generated class). But when the <i>jakarta.inject.Provider</i> type is used within your application then some of that deterministic behavior goes away. This is due to how Provider<>'s work since the implementation for the Provider (i.e., your application logic) "owns" the behavior for what actual concrete type that are created by it, along with the scope/cardinality for those instances. These instances are then delivered (as potentially injectable services) into other dependent services.  In this way Helidon Injection is simply acting as a broker and delivery mechanism between your Provider<> implementation(s) and the consumer that are using those service instances as injection points, etc. This is not meant to scare or even dissuade you from using Provider<>, but merely to inform you that some of the deterministic behavior goes away under these circumstances using Provider instead of another Scope type like @Singleton. In many/most cases this is completely normal and acceptable. As a precaution, however, Helidon Injection chose to fail-fast at application creation time if your application is found to use jakarta.inject.Provider<T>. You will then need to provide a strategy/configuration in your pom.xml file to permit these types of usages. There are options to allow ALL providers (as shown in the above example), or the strategy can be dictated on a case-by-case basis. See the javadoc for [AbstractApplicationCreatorMojo](src/main/java/io/helidon/inject/maven/plugin/AbstractApplicationCreatorMojo.java) for details.
```